src/remake.jl

```diff
@@ -99,6 +99,105 @@ function remake(
     _remake_internal(prob; kwargs..., p)
 end
 
+"""
+    $(TYPEDSIGNATURES)
+
+A utility function which merges two `NamedTuple`s `a` and `b`, assuming that the
+keys of `a` are a subset of those of `b`. Values in `b` take priority over those
+in `a`, except if they are `nothing`. Keys not present in `a` are assumed to have
+a value of `nothing`.
+"""
+function _similar_namedtuple_merge_ignore_nothing(a::NamedTuple, b::NamedTuple)
+    ks = fieldnames(typeof(b))
+    return NamedTuple{ks}(ntuple(Val(length(ks))) do i
+        something(get(b, ks[i], nothing), get(a, ks[i], nothing), Some(nothing))
+    end)
+end
+
```

Comment:
isn't this just `(;a..., b...)`?

Reply:
No, because `nothing` values in `b` will override values in `a`. This function makes it so that if a key has the value `nothing` in `b`, it uses the value from `a`.